### PR TITLE
IGAPP-494: Setup CI for e2e tests

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: .circleci/src/workflows/commit.yml [hunk 1]
```diff
@@ -8,6 +8,12 @@ jobs:
   - check_web:
       context:
         - mattermost
+  - e2e_web:
```
Comment (klinzo) [inline .circleci/src/workflows/commit.yml:11]: Shouldn't we execute this for other workflows as well?

Comment (klinzo) [reply .circleci/src/workflows/commit.yml:11]: If you did not this anywhere else because no tests for web are run there already, I think it would be good to add a todo there/add this as well but as comments.

File: build-configs/integreat-e2e/assets/app-icon-inverted.svg [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="59" height="59" viewBox="0 0 59 59">
```
Comment (klinzo) [inline build-configs/integreat-e2e/assets/app-icon-inverted.svg:1]: What's up with these new assets? Why are those needed explicitly?

File: build-configs/integreat-e2e/index.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import integreatTestCmsPlatformBuildConfigs from '../integreat-test-cms'
+import integreatPlatformBuildConfigs from '../integreat'
```
Comment (klinzo) [inline build-configs/integreat-e2e/index.ts:1]: Why do we now use integreat as base?

File: e2e-tests/web/wdio.conf.ts [hunk 4]
```diff
@@ -1,36 +1,30 @@
...
 
-  onPrepare: async function (): Promise<void> {
-    const startupDelay = 20000
-    await new Promise(resolve => setTimeout(resolve, startupDelay))
-  },
+
 
   before: async function (): Promise<void> {
-    await browser.setTimeout({ implicit: 80000 })
+    await browser.setTimeout({ implicit: 80000, pageLoad: 40000 })
```
Comment (klinzo) [inline e2e-tests/web/wdio.conf.ts:28]: Did you add this because of webpack startup times?

File: e2e-tests/web/capabilities.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    browserName: 'Firefox',
+    browserVersion: '84.0'
+  }),
+  safari: browserstackCaps({
+    os: 'OS X',
+    osVersion: 'Big Sur',
+    browserName: 'Safari',
+    browserVersion: '14.0'
+  }),
+  windows: browserstackCaps({
```
Comment (klinzo) [inline e2e-tests/web/capabilities.ts:38]: I think this can be removed as we won't support IE11 in the future.

**Comment by klinzo** [comment]
Looks very clean! :tada:

File: build-configs/integreat-e2e/index.ts (hunk 3 above)
Comment (Taggotty) [reply build-configs/integreat-e2e/index.ts:1]: The testumgebung-e2e instance is only on the live cms.

File: e2e-tests/web/wdio.conf.ts (hunk 4 above)
Comment (Taggotty) [reply e2e-tests/web/wdio.conf.ts:28]: yep. Will do the same thing as the onPrepare hook. And and also increases the general page loading time limits

File: build-configs/integreat-e2e/assets/app-icon-inverted.svg (hunk 2 above)
Comment (Taggotty) [reply build-configs/integreat-e2e/assets/app-icon-inverted.svg:1]: Those are basically just a copy of the integreat-test-cms assets, as those are needed to start the live cms app. Maybe you know a solution to reference the assets from the build config?

File: e2e-tests/web/wdio.conf.ts (hunk 4 above)
Comment (Taggotty) [reply e2e-tests/web/wdio.conf.ts:33]: Correction: I readded the initial startup delay, as the webapp takes about 5 seconds until the packager started the server. I still think the pageLoad is worth it to avoid failing tests due to web app instability

File: build-configs/integreat-e2e/assets/app-icon-inverted.svg (hunk 2 above)
Comment (klinzo) [reply build-configs/integreat-e2e/assets/app-icon-inverted.svg:1]: I think symlinking should work, didn't we use that before already?

File: build-configs/integreat-e2e/index.ts (hunk 3 above)
Comment (klinzo) [reply build-configs/integreat-e2e/index.ts:1]: There are other build config settings that differ between integreat and integreat-test-cms, did you check whether that are all okay?

**Review by SteffiStuffel — APPROVED** [review]
Really cool. :tada: 

just one small thing: the PR has the wrong prefix.

File: build-configs/integreat-e2e/index.ts (hunk 3 above)
Comment (Taggotty) [reply build-configs/integreat-e2e/index.ts:1]: for web I don't think there will be any problem. But can you check the app settings? Not sure whether the google stuff messes with anything.

File: docs/e2e-tests.md [hunk 6]
```diff
@@ -59,8 +29,38 @@ The [E2E test folder](../e2e-tests/web/test) contains two important subdirectori
...
+yarn workspace e2e prepare:android
+yarn workspace e2e prepare:start
+```
+
+It is recommended to start the app once manually to avoid timeouts during local testing. After you have the app installed and running you can execute the native E2E-tests with
+
+```
+yarn workspace e2e test:native
+```
+
```
Comment (klinzo) [inline docs/e2e-tests.md:62]:
```suggestion
```

File: docs/e2e-tests.md [hunk 7]
```diff
@@ -59,8 +29,38 @@ The [E2E test folder](../e2e-tests/web/test) contains two important subdirectori
...
+To run the web E2E-tests you first have to start the web app, and then execute the e2e tests:
+
+```
+yarn workspace e2e prepare:web:start
+yarn workspace e2e test:web
+```
+
+This will launch an automated chrome session where you can follow the test execution. If your don't have chrome installed you can adjust the `browserName` (firefox, edge, safari) in the `wdio.conf.ts` in the E2E-test web subdirectory.
+
+
```
Comment (klinzo) [inline docs/e2e-tests.md:45]:
```suggestion
```

File: docs/e2e-tests.md [hunk 8]
```diff
@@ -59,8 +29,38 @@ The [E2E test folder](../e2e-tests/web/test) contains two important subdirectori
...
+### Local Testing
+
+To run the web E2E-tests you first have to start the web app, and then execute the e2e tests:
+
+```
+yarn workspace e2e prepare:web:start
+yarn workspace e2e test:web
+```
+
+This will launch an automated chrome session where you can follow the test execution. If your don't have chrome installed you can adjust the `browserName` (firefox, edge, safari) in the `wdio.conf.ts` in the E2E-test web subdirectory.
```
Comment (klinzo) [inline docs/e2e-tests.md:43]: You could link the wdio conf here

Comment (klinzo) [reply docs/e2e-tests.md:43]: Could you mention that chromium does not work/only work after changing the browser name?

**Review by klinzo — APPROVED** [review]

**Review by f1sh1918 — APPROVED** [review]
nice work :)
